### PR TITLE
ARTEMIS-5182 STOMP subscriber w/noLocal + selector missing messages

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

File: artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompConnection.java [hunk 1]
```diff
@@ -566,7 +566,7 @@ StompPostReceiptFunction subscribe(String destination,
          if (selector == null) {
             selector = noLocalFilter;
          } else {
-            selector += " AND " + noLocalFilter;
+            selector = "(" + selector + ") AND " + noLocalFilter;
```
Comment (brusdev) [inline artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompConnection.java:569]: could selector be an empty string?

Comment (jbertram) [reply artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompConnection.java:569]: Yes, I believe so. I'll add some validation.

Comment (jbertram) [reply artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompConnection.java:572]: Validation added.

**Review by brusdev — APPROVED** [review]
Full test-suite and example passed

**Pull request merged**